cpp/src/slice2matlab/Main.cpp

```diff
@@ -1598,30 +1679,47 @@ CodeVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
     out << nl << "function r = checkedCast(p, varargin)";
     out.inc();
     out << nl << "% checkedCast   Contacts the remote server to verify that the object implements this type.";
-    out << nl << "%   Raises a local exception if a communication error occurs. You can optionally supply a";
-    out << nl << "%   facet name and a context map.";
     out << nl << "%";
     out << nl << "% Parameters:";
-    out << nl << "%   p - The proxy to be cast.";
-    out << nl << "%   facet - The optional name of the desired facet.";
-    out << nl << "%   context - The optional context map to send with the invocation.";
+    out << nl << "%   p - The proxy to be check.";
+    out << nl << "%   facet - The desired facet (optional).";
+    out << nl << "%   context - The request context (optional).";
     out << nl << "%";
     out << nl << "% Returns (" << prxAbs << ") - A proxy for this type, or an empty array if the object"
         << " does not support this type.";
+    out << nl << "arguments";
+    out.inc();
+    out << nl << "p Ice.ObjectPrx {mustBeScalarOrEmpty}";
+    out.dec();
+    out << nl << "end";
+    out << nl << "arguments (Repeating)";
+    out.inc();
+    out << nl << "varargin % facet or context, or both, or neither";
```

Comment:
Using varargin in `arguments` is not recommended:
https://www.mathworks.com/help/matlab/matlab_prog/validate-repeating-arguments.html#mw_7ef08fa3-36ed-47f1-a967-1697e3e79d2c

We fall in the "support legacy code" exception.